.vscode/settings.json

```diff
@@ -1,5 +1,4 @@
 {
-  "editor.formatOnSave": true,
```

Comment:
We do want to have this flag set. Otherwise, people will get surprise CI build failures.

Reply:
How is this property used during CI?   With it on, hitting save in vs code causes the whole file to get reformatted in a way that is inconsistent with all the other files in the repo.

Reply:
We run a `lint` rule as part of CI. If someone make changes but don't format it according to the lint rules, the build fails. 

I suspect the reason that VS formatting doesn't match the other files is that we turned on this `formatOnSave` only recently and many files were already checked in before that.

Reply:
Interesting.  The `lint` succeeds with the current formatting or the settings.json formatting.   I reverted the change

Reply:
For a while we had a few settings undefined which was causing different editors to save in different settings. I believe we've now defined those values such that everyone should be using the same settings. This means that as people edit the files in the repo they will be converted to be consistent one by one.

Generally the formatter doesn't run as part of a CI unless it gets run as part of something else. If that happens it could break CI in that it will prompt that there are unsubmitted changes. (This would be like running yarn causing the lock file to be updated, this would trigger the same check)